lib/streamlit/delta_generator.py

```diff
@@ -509,6 +520,20 @@ def _enqueue(
         """
         # Operate on the active DeltaGenerator, in case we're in a `with` block.
         dg = self._active_dg
+
+        ctx = get_script_run_ctx()
+        if (
+            ctx
+            and ctx.fragment_ids_this_run
+            and ctx.current_fragment_id in ctx.fragment_ids_this_run
+            and _writes_directly_to_sidebar(dg)
+        ):
+            raise StreamlitAPIException(
+                "Adding elements directly to `st.sidebar` from a function wrapped with "
+                "`st.experimental_fragment` is not supported. To execute fragment code "
+                "in the sidebar, use `with st.sidebar: fragment_func()` or similar."
```

Comment:
WDYT about Debbie's version from the docstring? Even clearer, I think.

"Calling `st.sidebar` in a function wrapped with `st.experimental_fragment` is not supported. To write elements to the sidebar with a fragment, call your fragment function inside a `with st.sidebar` context manager."

Reply:
Ah, yeah -- that is clearer. Changed this message to match the docstring